modules/cudafeatures2d/test/test_features2d.cpp

```diff
@@ -102,6 +108,62 @@ CUDA_TEST_P(FAST, Accuracy)
     }
 }
 
+#ifdef _OPENMP
+CUDA_TEST_P(FAST, Async)
+{
+    cv::Mat image_ = readImage("features2d/aloe.png", cv::IMREAD_GRAYSCALE);
+    ASSERT_FALSE(image_.empty());
+
+    if (!supportFeature(devInfo, cv::cuda::GLOBAL_ATOMICS))
+    {
+        try
+        {
+            cv::Ptr<cv::cuda::FastFeatureDetector> fast = cv::cuda::FastFeatureDetector::create(threshold, nonmaxSuppression);
+            std::vector<cv::KeyPoint> keypoints;
+            fast->detect(loadMat(image_), keypoints);
+        }
+        catch (const cv::Exception& e)
+        {
+            ASSERT_EQ(cv::Error::StsNotImplemented, e.code);
+        }
+    }
+    else
+    {
+        cv::cuda::HostMem image(image_);
+
+        GpuMat d_keypoints[2];
+        std::vector<cv::KeyPoint> keypoints[2];
+        cv::Ptr<cv::cuda::FastFeatureDetector> d_fast[2];
+
+        d_fast[0] = cv::cuda::FastFeatureDetector::create(threshold, nonmaxSuppression);
+        d_fast[1] = cv::cuda::FastFeatureDetector::create(threshold, nonmaxSuppression);
+
+        #pragma omp parallel num_threads(2)
```

Comment:
Please use `cv::parallel_for_(Range(0, 2), ...)` instead of OpenMP.

[Example](https://github.com/opencv/opencv/blob/3.4.0/modules/core/test/test_rand.cpp#L386-L420)

Reply:
Ok, I'll fix it in a few hours.